### PR TITLE
Store contained level responses in sessionStorage

### PR DESCRIPTION
**Pull request by balderdash**
This makes contained levels work for non-signed-in students (until they clear their session storage at least), i.e. when they go back to a level for which they'd already submitted a response, their previous response shows up but can't be edited, and the run button starts out enabled.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
Changes look good. Would be great to get some test coverage of this new behavior, though I'm not sure what the best way of accomplishing that is.

**Comment by balderdash** [comment]
I'll see what I can do.

**Comment by balderdash** [comment]
PTAL

Wrote some unit tests for the freeResponse.js and multi.js changes. Writing unit tests for loadApp.js didn't go so well.

I'll add an eyes test for CSF contained levels in another PR.

File: apps/test/unit/code-studio/levels/freeResponse.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+describe('Free Response', () => {
+  const levelId = 1047;
+  const scriptName = 'test-script';
+  const lastAttemptString = 'This is my final answer';
+  const otherLastAttemptString = 'This is some other answer';
+  let textarea;
+
+  before(() => {
+    textarea = document.createElement('textarea');
```
Comment (Bjvanminnen) [inline apps/test/unit/code-studio/levels/freeResponse.js:16]: Might add a short comment explaining why we're creating this text area/what it is.

File: apps/test/unit/code-studio/levels/freeResponse.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    textarea.setAttribute('id', `level_${levelId}`);
+    textarea.setAttribute('class', 'response');
+    document.body.appendChild(textarea);
+    window.appOptions = {};
```
Comment (Bjvanminnen) [inline apps/test/unit/code-studio/levels/freeResponse.js:22]: I might stash the currently window.appOptions, and then reset it (like here https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/code-studio/levels/multi.js#L6) to reduce the likelihood that changes to appOptions leak between tests.

**Pull request merged**